integration/integration_test.go

```diff
@@ -221,47 +185,42 @@ func TestRun(t *testing.T) {
 }
 
 func TestLayers(t *testing.T) {
-	dockerfiles, err := filepath.Glob(path.Join(dockerfilesPath, "Dockerfile_test*"))
-	if err != nil {
-		t.Error(err)
-		t.FailNow()
-	}
 	offset := map[string]int{
 		"Dockerfile_test_add":     9,
 		"Dockerfile_test_scratch": 3,
 		// the Docker built image combined some of the dirs defined by separate VOLUME commands into one layer
 		// which is why this offset exists
 		"Dockerfile_test_volume": 1,
 	}
-	for _, dockerfile := range dockerfiles {
+	for dockerfile, built := range imageBuilder.FilesBuilt {
 		t.Run("test_layer_"+dockerfile, func(t *testing.T) {
-			dockerfile = dockerfile[len("dockerfile/")+1:]
-			for _, ignore := range testsToIgnore {
-				if dockerfile == ignore {
-					t.SkipNow()
+			if !built {
+				err := imageBuilder.BuildImage(config.imageRepo, config.gcsBucket, dockerfilesPath, dockerfile)
+				if err != nil {
+					t.Fatalf("Failed to build kaniko and docker images for %s: %s", dockerfile, err)
 				}
 			}
 			// Pull the kaniko image
-			dockerImage := strings.ToLower(testRepo + dockerPrefix + dockerfile)
-			kanikoImage := strings.ToLower(testRepo + kanikoPrefix + dockerfile)
+			dockerImage := GetDockerImage(config.imageRepo, dockerfile)
+			kanikoImage := GetKanikoImage(config.imageRepo, dockerfile)
 			pullCmd := exec.Command("docker", "pull", kanikoImage)
 			RunCommand(pullCmd, t)
-			if err := checkLayers(dockerImage, kanikoImage, offset[dockerfile]); err != nil {
+			if err := checkLayers(t, dockerImage, kanikoImage, offset[dockerfile]); err != nil {
 				t.Error(err)
 				t.Fail()
 			}
 		})
 	}
 }
 
-func checkLayers(image1, image2 string, offset int) error {
+func checkLayers(t *testing.T, image1, image2 string, offset int) error {
```

Comment:
Why do we need to pass t in here?

Reply:
oh thanks for catching that @sharifelgamal I'll remove it - I had tried out calling `t.Helper()` in this function to change the line number reported for failures, but I ended up thinking it's better reporting the line number inside `checkLayers`